CMake error when the source directory full path contains "C++/"

_Please use the following template to help us solving your issue._

## Issue Details

Error when running cmake, with examples ON.

## Source Code

cgal master

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Ubuntu 20.04.6 LTS
* CGAL version: master branch
* cmake version: 3.27.8

Here is the first error: 
````
== Generating build files for examples ==
CMake Error at Installation/cmake/modules/CGAL_Macros.cmake:587 (string):
  string sub-command REGEX, mode REPLACE failed to compile regex
  "/home/to/dev/C++/cgal2/.*/examples/".
Call Stack (most recent call first):
  Installation/examples/CMakeLists.txt:32 (process_cgal_subdirectory)

````

--- a/Installation/cmake/modules/CGAL_Macros.cmake
+++ b/Installation/cmake/modules/CGAL_Macros.cmake
@@ -579,15 +579,9 @@ if( NOT CGAL_MACROS_FILE_INCLUDED )
 
 endif()
 
-
 function(process_CGAL_subdirectory entry subdir type_name)
   # For example, subdir can be "examples", type_name "example", and entry "Mesh_2"
-
-  if ( CGAL_BRANCH_BUILD )
-    string( REGEX REPLACE "${CMAKE_SOURCE_DIR}/.*/${subdir}/" "" ENTRY_DIR_NAME "${entry}" )
-  else()
-    string( REGEX REPLACE "${CMAKE_CURRENT_SOURCE_DIR}/" "" ENTRY_DIR_NAME "${entry}" )
-  endif()
+  get_filename_component(ENTRY_DIR_NAME "${entry}" NAME)
 
   if( NOT "${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}") # out-of-source
     make_directory("${CMAKE_BINARY_DIR}/${subdir}/${ENTRY_DIR_NAME}")

